src/components/Integrations/Shopify/Shopify.js

```diff
@@ -68,24 +70,46 @@ const Shopify = ({ intl, dependencies: { shopifyClient } }) => {
       </tbody>
     </table>
   );
+
+  function useInterval(callback, delay) {
+    const savedCallback = useRef();
+
+    useEffect(() => {
+      savedCallback.current = callback;
+    }, [callback]);
+
+    useEffect(() => {
+      function tick() {
+        savedCallback.current();
+      }
+      if (delay !== null) {
+        let id = setInterval(tick, delay);
+        return () => clearInterval(id);
+      }
+    }, [delay]);
+  }
+
   useEffect(() => {
     const getData = async () => {
       const result = await shopifyClient.getShopifyData();
       if (!result.success) {
         setError(<FormattedHTMLMessage id="validation_messages.error_unexpected_HTML" />);
       } else if (result.value.length) {
         setIsConnected(true);
+        setDoPoll(0);
```

Comment:
it's not better use and object and update component state only one time and don't four times?

Reply:
An object with what exactly? because I don't use all variables all the time...

Reply:
What @fcoronelmakingsense said is reasonable: if a set of values change always together, it is useful to store the in only one state variable. 

Probably React has optimizations to avoid renders when the state will change soon, but, we could help and also the code could be cleaner.

@fcoronelmakingsense you have applied that approach in another component could you link it in this conversation?

Reply:
i apply in `ReportsTrafficsources`, you can check this component @cbernat

Reply:
Here you are:

https://github.com/FromDoppler/doppler-webapp/blob/58986eaeb82e3d02714f8994949c4f236a207e40/src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js#L35-L41

---

BTW, see more about this GitHub feature at:

* https://github.blog/2017-08-15-introducing-embedded-code-snippets/ 
* https://help.github.com/en/articles/creating-a-permanent-link-to-a-code-snippet

![image](https://github.blog/wp-content/uploads/2017/08/29129733-c682da44-7cf5-11e7-8c73-7eaea274c5ae.gif)

Reply:
Fixed.